test/Base_Internal_Tests/src/Types/Any_Intersection_Spec.enso

```diff
@@ -0,0 +1,46 @@
+from Standard.Base import all
+from Standard.Test import all
+import Standard.Base.Internal.Meta_Helpers
+
+from project.Types.Any_Intersection import A, B, C, abc, ac_hidden_b
+
+add_specs suite_builder =
+    suite_builder.group "Any Intersection Tests" group_builder->
+        type_name = Meta_Helpers.simple_type_name_builtin
+
+        extract_b_1 b:B -> B = b
+        extract_b_2 b -> B = b
+        extract_b_3 b = b:B
+        extract_b_4 b:B = b
+
+        group_builder.specify "Extract B visible type" <|
+            v = abc
+            type_name v . should_equal "A & B & C"
+
+            r1 = extract_b_1 v
+            r2 = extract_b_2 v
+            r3 = extract_b_3 v
+            r4 = extract_b_4 v
+
+            type_name r1 . should_equal "B & ~A & ~C"
+            type_name r2 . should_equal "B & ~A & ~C"
+            type_name r3 . should_equal "B & ~A & ~C"
+            type_name r4 . should_equal "B & ~A & ~C"
+
+        group_builder.specify "Extract B hidden type" <|
+            v = ac_hidden_b
+
+            r1 = Panic.recover Any <| extract_b_1 v
+            r2 = extract_b_2 v
+            r3 = extract_b_3 v
+            r4 = Panic.recover Any <| extract_b_4 v
+
+            type_name r1 . should_equal "Error"
+            type_name r2 . should_equal "B & ~A & ~C"
```

Comment:
- this assert shows the specification written in 3cc868f69d4f9aa24ef9de8d88aed12af9d4f482 is violated
- it shows that `-> B`  is actually _searching hidden types_
- that signals the #12292 report isn't off
- clearly `->` conversions are converting a _"bit too much"_

Reply:
As of 1f2efb7b9d2edb7b389ee20bf51e68926eeda15c all the problems are addressed.